openmdao/solvers/linesearch/tests/test_backtracking.py

```diff
@@ -254,12 +254,60 @@ def linearize(self, inputs, outputs, jac):
         jac[('z', 'z')] = -inputs['x'] + 1.0
         jac[('z', 'x')] = -outputs['z']
 
+class ImplCompTwoStatesAEBug(om.ImplicitComponent):
```

Comment:
If you just set new counter limits in `ImplCompTwoStatesAE`, do you get a test failure? If you don't, then you don't need the extra class.  If you do, then you can still make this simpler by inheriting from `ImplCompTwoStatesGuess` and just replacing the apply_nonlinear function with the new one that has new limits.  (You could also do it in one class if you let the upper and lower check numbers be set as an attribute.